tree/binary.go

```diff
@@ -17,6 +17,10 @@ type (
 		left  *node
 		right *node
 	}
+	lvlNode struct {
```

Comment:
This can be inside the function as local type as we are not using it somewhere else. If there will be requirement for it's reuse we can then create this as a package level type.